lib/base-compiler.js

```diff
@@ -291,23 +297,31 @@ Compile.prototype.compile = function (source, options, backendOptions, filters)
                 }
                 return result;
             })
-            .then(_.bind(self.postProcessAsm, self))
+            .then(_.bind(function(result){return self.postProcessAsm(result, filters);}, self))
             .then(function (result) {
                 if (result.okToCache) self.env.cachePut(key, result);
                 return result;
             });
     });
 };
 
-Compile.prototype.postProcessAsm = function (result) {
+Compile.prototype.postProcessAsm = function (result, filters) {
+    var self = this;
     if (!result.okToCache) return result;
-    var demangler = this.compiler.demangler;
+    var demangler = self.compiler.demangler;
     if (!demangler) return result;
-    return this.exec(demangler, [], {input: _.pluck(result.asm, 'text').join("\n")})
+    return self.exec(demangler, [], {input: _.pluck(result.asm, 'text').join("\n")})
         .then(function (demangleResult) {
             var lines = utils.splitLines(demangleResult.stdout);
             for (var i = 0; i < result.asm.length; ++i)
                 result.asm[i].text = lines[i];
+            result.supportCfg = false;
+            if(result.code === 0 && self.supportCFGGeneration(self.compiler.version) &&
+                    filters.commentOnly && filters.directives && filters.labels){
```

Comment:
Why is the CFG only enabled when the filters are in this mode? I realise 99.99% of Compiler Explorer users will use these filters in their default state but it seems like an unnecessary limitation, and potentially confusing to the user if they toggle the "directives filter" button and lose their CFG.

Reply:
discussed in slack.